File: code/packages/ruby/paint_codec_png_native/test/paint_codec_png_native_test.rb
# frozen_string_literal: true

require_relative "test_helper"
require "coding_adventures/pixel_container"

class PaintCodecPngNativeTest < Minitest::Test
  def test_available
    assert CodingAdventures::PaintCodecPngNative.available?
  end

  def test_encode_png
    pixels = CodingAdventures::PixelContainer::Container.new(1, 1, [0, 0, 0, 255].pack("C*").b)
    png = CodingAdventures::PaintCodecPngNative.encode(pixels)
    assert_equal "\x89PNG\r\n\x1a\n".b, png.byteslice(0, 8)
  end
end

File: code/packages/elixir/paint_codec_png_native/test/paint_codec_png_native_test.exs
defmodule CodingAdventures.PaintCodecPngNativeTest do
  use ExUnit.Case, async: false

  alias CodingAdventures.{PaintCodecPngNative, PixelContainer}

  test "encodes RGBA pixels to a PNG binary" do
    pixels = %PixelContainer{
      width: 1,
      height: 1,
      data: <<0, 0, 0, 255>>
    }

    assert PaintCodecPngNative.available?()
    assert {:ok, png} = PaintCodecPngNative.encode(pixels)
    assert binary_part(png, 0, 8) == <<137, 80, 78, 71, 13, 10, 26, 10>>
  end
end
